frontend/pages/reports/index.js

```diff
@@ -0,0 +1,227 @@
+import Edit from './edit'
+
+const GroupLabel = s.h4`
+  margin-bottom: 0em !important;
+  font-size: 32px;
+  color: #626572;
+`
+
+const ColoredHeader = s.div`
+  background: linear-gradient(to right, #ef4c5f, #4954f4);
+  height: 7em;
+`
+
+const TransparentButton = s.button`
+  width: 190px;
+  height: 35px;
+  border-radius: 17px;
+  border: 0;
+  background: rgba(255,255,255,0.32);
+  float: right;
+  font-size: 15px;
+  font-family: Lato;
+  font-weight: 500;
+  font-stretch: normal;
+  font-style: normal;
+  line-height: normal;
+  letter-spacing: normal;
+  text-align: center;
+  color: rgba(255,255,255,1)
+`
+
+const serializeParams = params => {
+  return Object.keys(params)
+    .map(key => `${encodeURIComponent(key)}=${encodeURIComponent(params[key])}`)
+    .join('&')
+}
+
+const Reports = ({ nameToCode }) => {
+  const fields = {
+    Fields: Object.keys(nameToCode),
+  }
+
+  const [reports, setReports] = useState([])
+  const [reportFlag, updateReportFlag] = useState(false)
+
+  const [isEdit, setIsEdit] = useState(false)
+
+  useEffect(() => {
+    doApiRequest('/reports/?format=json')
+      .then(resp => {
+        if (resp.ok) {
+          return resp.json()
+        } else {
+          return []
+        }
+      })
+      .then(data => setReports(data))
+  }, [reportFlag])
+
+  const [includedFields, setIncludedFields] = useState(
+    (() => {
+      const initial = {}
+      Object.keys(fields).forEach(group =>
+        fields[group].forEach(f => {
+          initial[f] = false
+        })
+      )
+      return initial
+    })()
+  )
+
+  const query = {
+    format: 'xlsx',
+    fields: Object.keys(includedFields)
+      .filter(field => includedFields[field])
+      .map(name => nameToCode[name])
+      .filter(e => e !== undefined),
+  }
+
+  const generateCheckboxGroup = (groupName, fields) => {
+    return (
+      <div key={groupName} style={{ flexBasis: '50%', flexShrink: 0 }}>
+        <GroupLabel
+          key={groupName}
+          className="subtitle is-4"
+          style={{ color: CLUBS_GREY }}
+        >
+          {groupName}
+        </GroupLabel>
+        {fields.map((field, idx) => (
+          <div key={idx}>
+            <Checkbox
+              id={field}
+              checked={includedFields[field]}
+              onChange={() => {
+                setIncludedFields(prev => ({ ...prev, [field]: !prev[field] }))
+              }}
+            />
+            {'  '}
+            <CheckboxLabel htmlFor={field}>{field}</CheckboxLabel>
+          </div>
+        ))}
+      </div>
+    )
+  }
+
+  const handleBack = () => {
+    setIsEdit(false)
+    Router.push('/reports')
+  }
+
+  return (
+    <div>
+      {isEdit ? (
+        <div>
+          <ColoredHeader>
+            <div style={{ padding: '2em' }}>
+              <span className="title is-3" style={{ color: 'white' }}>
+                Create a new report
+              </span>
+              <TransparentButton onClick={() => handleBack()}>
```

Comment:
could just be onClick={handleBack} I think, unless there's some reason you want to call handleBack with no args here?

Reply:
I've found in eslint things that this syntax is actually preferred. When/if we move to TypeScript the linter will prefer we use this style (deliberately not sending the event if we don't need it)

Reply:
makes sense, () => handleBack() swallows the event so it makes the types nicer, and if handleBack checks to see if any meaningful params are passed in this lets handleBack know there isn't.